test/multi_instance_test.go

```diff
@@ -1,103 +1,127 @@
-}
-
-func TestMultiASyncPair(t *testing.T) {
-	sm, err := sim.NewSimulation(asyncOptions(t, 1413,
-		sim.AddHonestParticipants(2, sim.NewUniformECChainGenerator(tipSetGeneratorSeed, 1, 10), uniformOneStoragePower),
-	)...)
-	require.NoError(t, err)
-
-	require.NoErrorf(t, sm.Run(instanceCount, maxRounds), "%s", sm.Describe())
-	// Note: when async, the decision is not always the latest possible value,
-	// but should be something recent.
-	// This expectation may need to be relaxed.
-	instance := sm.GetInstance(instanceCount)
-	require.NotNil(t, instance)
-	expected := instance.BaseChain
-	requireConsensusAtInstance(t, sm, instanceCount-1, expected...)
-}
+// TestHonestMultiInstance_Agreement tests for multiple chained instances of the protocol with no adversaries.
+func TestHonestMultiInstance_Agreement(t *testing.T) {
+	t.Parallel()
+	const (
+		instanceCount            = 4000
+		testRNGSeed              = 8965130
+		latencySeed              = testRNGSeed * 7
+		maxParticipantCases      = 20
+		maxParticipantCasesShort = maxParticipantCases / 2
+	)
 
-func TestMultiSyncAgreement(t *testing.T) {
+	// The number of honest participants for which every table test is executed.
+	var participantCounts []int
 	if testing.Short() {
-		t.Skip("too slow for testing.Short")
+		// Reduce max participants to 5 during short test for faster completion.
+		participantCounts = make([]int, maxParticipantCasesShort)
+	} else {
+		participantCounts = make([]int, maxParticipantCases)
+	}
+	for i := range participantCounts {
+		participantCounts[i] = i + 1
 	}
 
-	t.Parallel()
-	repeatInParallel(t, 9, func(t *testing.T, repetition int) {
-		honestCount := repetition + 3
-		sm, err := sim.NewSimulation(syncOptions(
-			sim.AddHonestParticipants(
-				honestCount,
-				sim.NewUniformECChainGenerator(tipSetGeneratorSeed, 1, 10),
-				uniformOneStoragePower),
+	tests := []struct {
+		name    string
+		options []sim.Option
+	}{
+		{
+			name:    "sync",
+			options: syncOptions(),
+		},
+		{
+			name:    "async",
+			options: asyncOptions(latencySeed),
+		},
+	}
+	for _, participantCount := range participantCounts {
+		participantCount := participantCount
+		for _, test := range tests {
+			test := test
+			name := fmt.Sprintf("%s %d", test.name, participantCount)
+			t.Run(name, func(t *testing.T) {
+				multiAgreementTest(t, testRNGSeed, participantCount, instanceCount, test.options...)
+			})
+		}
+	}
+}
+
+// FuzzHonestMultiInstance_AsyncDisagreement tests a scenario where two groups of equal
+// participants, both in terms of count and power, vote for randomly generated
+// chains at each instance, where the chain is uniform among all participants
+// within the same group. It then asserts that over multiple instances the
+// consensus is never reached and all nodes converge on the base chain of the
+// first instance.
+func FuzzHonestMultiInstance_AsyncDisagreement(f *testing.F) {
+	const (
+		instanceCount = 1000
+		honestCount   = 20
```

Comment:
I think you're still WIP here, again here I'd suggest trying a range of counts, and smaller counts mostly more valuable than larger ones (also faster!).

Ditto below